mongo/tests/test_unit.py

```diff
@@ -601,3 +595,14 @@ def test_when_version_lower_than_3_6_then_no_session_metrics_reported(aggregator
         dd_run_check(check)
     # Then
     aggregator.assert_metric('mongodb.sessions.count', count=0)
+
+
+def test_service_check_critical_when_connection_dies(aggregator, check, instance, dd_run_check):
+    check = check(instance)
+    with mock_pymongo('standalone') as mocked_client:
+        dd_run_check(check)
+        aggregator.assert_service_check('mongodb.can_connect', MongoDb.OK)
+        aggregator.reset()
+        mocked_client.list_database_names = mock.MagicMock(side_effect=ConnectionFailure("Testing"))
+        dd_run_check(check)
+        aggregator.assert_service_check('mongodb.can_connect', MongoDb.CRITICAL)
```

Comment:
we might parametrize this test to test the three different type of errors that can come from the mongo client

Reply:
Good idea! I'll do that locally!